pkg/envoy/envoy_version_sha256.go

```diff
@@ -0,0 +1,11 @@
+// Code generated by generate-cilium-envoy-version-sha256. DO NOT EDIT.
+
+// Copyright 2021 Authors of Cilium
+// SPDX-License-Identifier: Apache-2.0
+
+package envoy
+
+// RequiredEnvoyVersionSHA is set during build
+// Running Envoy version will be checked against `RequiredEnvoyVersionSHA`.
+// By default cilium-agent will fail to start if there is a version mismatch.
+const RequiredEnvoyVersionSHA = "3ae9ae3ca32eb30816c9e6a655dcb36e7ba4183c8c0c93ad58b72ad6c064aa0b"
```

Comment:
If we're committing these to the tree now, does this cause weirdness with git any time we build? Should we add these files to .gitignore?

Also, do we need to update these SHAs for each release or anything like that? Or are these just dummy values? In which case should we make them clearly dummy values using a magic pattern?

Reply:
These files should not be in `.gitignore` as they are needed for Cilium to build from a clean checkout.

> Also, do we need to update these SHAs for each release or anything like that?

The SHA256s should be updated automatically whenever you run `make`. As the generated files are checked into git, any changes to them will be visible in `git diff` and should be checked in alongside the commit that causes the SHA256 to change.

Reply:
> These files should not be in .gitignore as they are needed for Cilium to build from a clean checkout.

Would adding them to the tree + adding to .gitignore solve this? I understand that we need it available to be able to build, but it sounds like if the build is working correctly then any time we change bpf/ code, this file will change; but we don't really want to track the changes to the file.

> As the generated files are checked into git, any changes to them will be visible in git diff and should be checked in alongside the commit that causes the SHA256 to change.

We used to have a system like this in the past, it's a pain to maintain. This means if there are ever parallel BPF changes in flight, we are forced to rebase BPF PRs purely to satisfy this constraint :/

Reply:
^^ I think we should raise awareness of this proposal for this last point and see if we can come up with a better solution.